File: LeetCode/Easy/0496-next-greater-element-i/0496-next-greater-element-i.cpp
class Solution {
public:
    vector<int> nextGreaterElement(vector<int>& nums1, vector<int>& nums2) {
        int findBig[10000]{[0 ... 9999] = -1};
        stack<int> stack;
        vector<int> ret;

        for (int i = 0; i < nums2.size(); ++i){
            while(!stack.empty() && stack.top() < nums2[i]){
                int key = stack.top();
                findBig[key] = nums2[i];
                stack.pop();
            }
            stack.push(nums2[i]);
        }
        
        for (int i = 0; i < nums1.size(); ++i) ret.push_back(findBig[nums1[i]]);
        
        return ret;
    }
};

File: LeetCode/Easy/0496-next-greater-element-i/0496-next-greater-element-i.go
func nextGreaterElement(nums1 []int, nums2 []int) []int {
    var findBig [10000]int = [10000]int{0}
    ret := make([]int, len(nums1))
    stack := make([]int, 0)
    
    for _, v := range nums2 {
        for len(stack) != 0 && stack[len(stack)-1] < v {
            key := stack[len(stack)-1]
            findBig[key] = v
            stack = stack[:len(stack)-1]
        }
        stack = append(stack, v)
    }
    
    for i := 0; i < len(nums1); i++ {
        if findBig[nums1[i]] == 0 {
            ret[i] = -1
        } else {
            ret[i] = findBig[nums1[i]]
        }
    }
    
    return ret
}